hera_cal/lstbin.py

```diff
@@ -1000,22 +1002,24 @@ def sigma_clip(array, flags=None, sigma=4.0, axis=0, min_N=4):
     # create empty clip_flags array
     clip_flags = np.zeros_like(array, np.bool)
 
+    # make copy of data array as to not replace original array
+    array_ = array.copy()
+
     # inherit flags if fed and apply flags to data
     if flags is not None:
         clip_flags += flags
-        array[flags] *= np.nan
+        array_[flags] *= np.nan
 
     # get robust location
-    location = np.nanmedian(array, axis=axis)
+    location = np.nanmedian(array_, axis=axis)
 
     # get MAD! * 1.482579
-    scale = np.nanmedian(np.abs(array - location), axis=axis) * 1.482579
+    scale = np.nanmedian(np.abs(array_ - location), axis=axis) * 1.482579
 
     # get clipped data
-    clip = np.abs(array - location) / scale > sigma
+    clip = np.abs(array_ - location) / scale > sigma
 
-    # set clipped data to nan and set clipped flags to True
-    array[clip] *= np.nan
```

Comment:
I've removed this as it modifies the input array, which means that the Re and Im clipping may see different data arrays, meaning that the ordering matters. We don't want this, therefore `array[clip] *= np.nan` is removed.

Reply:
I don' think it does: you can have a complex number as 1 + 1j*nan. b/c we do `sigma_clip(data.real, ...` and `sigma_clip(data.imag)` I don't think this is actually a concern

Reply:
I'm not sure I follow? If `array[clip] *= np.nan` is left in the `sigma_clip` function, then the data array is modified in place and the second sigma_clip call will see a different data array.

Reply:
Ah yes, sorry just got my head around this. Thanks for explaining!